pkg/controller/stack/flux.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2022, Pulumi Corporation.  All rights reserved.
+
+package stack
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha1"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/auto"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/pulumi/pulumi-kubernetes-operator/pkg/apis/pulumi/shared"
+)
+
+func (sess *reconcileStackSession) SetupWorkdirFromFluxSource(ctx context.Context, source unstructured.Unstructured, fluxSource *shared.FluxSource) (_commit string, retErr error) {
+	rootdir, err := os.MkdirTemp("", "pulumi_source")
+	if err != nil {
+		return "", fmt.Errorf("unable to create tmp directory for workspace: %w", err)
+	}
+	sess.rootDir = rootdir
+
+	defer func() {
+		if retErr != nil {
+			_ = os.RemoveAll(rootdir)
+			sess.rootDir = ""
+		}
+	}()
+
+	// this source artifact fetching code is based closely on
+	// https://github.com/fluxcd/kustomize-controller/blob/db3c321163522259595894ca6c19ed44a876976d/controllers/kustomization_controller.go#L529
+	artifactURL, ok, err := unstructured.NestedString(source.Object, "status", "artifact", "url")
+	if !ok || err != nil {
+		return "", errors.New("expected source to have .status.artifact.url, but it did not")
+	}
+
+	revision, ok, err := unstructured.NestedString(source.Object, "status", "artifact", "revision")
+	if !ok || err != nil {
+		return "", errors.New("did not find revision in .status.artifact")
+	}
+
+	checksum, ok, err := unstructured.NestedString(source.Object, "status", "artifact", "checksum")
+	if !ok || err != nil {
+		return "", errors.New("did not find revision in .status.artifact")
```

Comment:
```suggestion
		return "", errors.New("did not find checksum in .status.artifact")
```

Reply:
I made a little helper so that these are consistent (and also check that the value isn't an empty string).